src/Uuid/Uuid.php

```diff
@@ -0,0 +1,349 @@
+
+        return pack('H*', $uuid);
+    }
+
+    public static function uuid_unparse($uuidAsBinary)
+    {
+        if (!\is_string($uuidAsBinary)) {
+            trigger_error(sprintf("uuid_unparse() expects parameter 1 to be string, %s given", gettype($uuid)), E_USER_WARNING);
+
+            return null;
+        }
+
+        $dataAsArray = unpack('H*', $uuidAsBinary);
+        $data = $dataAsArray[1];
+
+        if (32 != \strlen($data)) {
+            return false;
+        }
+
+        $uuid = sprintf('%s-%s-%s-%s-%s',
+            substr($data, 0, 8),
+            substr($data, 8, 4),
+            substr($data, 12, 4),
+            substr($data, 16, 4),
+            substr($data, 20, 12)
+        );
+
+        if (null === self::uuid_parse_as_array($uuid)) {
+            return false;
+        }
+
+        return $uuid;
+    }
+
+    private static function uuid_generate_random()
+    {
+        return sprintf('%08x-%04x-%04x-%04x-%06x%06x',
+            // 32 bits for "time_low"
+            random_int(0, 0xffffffff),
+
+            // 16 bits for "time_mid"
+            random_int(0, 0xffff),
+
+            // 16 bits for "time_hi_and_version",
+            // four most significant bits holds version number 4
+            random_int(0, 0x0fff) | 0x4000,
+
+            // 16 bits:
+            // * 8 bits for "clk_seq_hi_res",
+            // * 8 bits for "clk_seq_low",
+            // two most significant bits holds zero and one for variant DCE1.1
+            random_int(0, 0x3fff) | 0x8000,
+
+            // 48 bits for "node"
+            random_int(0, 0xffffff),
+            random_int(0, 0xffffff)
+        );
+    }
+
+    /**
+     * @see http://tools.ietf.org/html/rfc4122#section-4.2.2
+     */
+    private static function uuid_generate_time()
+    {
+        // https://tools.ietf.org/html/rfc4122#section-4.1.4
+        // 0x01b21dd213814000 is the number of 100-ns intervals between the
+        // UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00.
+        $offset = 0x01b21dd213814000;
+        $timeOfDay = gettimeofday();
+        $time = ($timeOfDay['sec'] * 10000000) + ($timeOfDay['usec'] * 10) + $offset;
+
+        // https://tools.ietf.org/html/rfc4122#section-4.1.5
+        // We are using a random data for the sake of simplicity: since we are
+        // not able to get a super precise timeOfDay as a unique sequence
+        $clockSeq = random_int(0, 0x3fff);
+
+        if (\function_exists('apcu_fetch')) {
+            $node = apcu_fetch('__symfony_uuid_node');
+            if (false === $node) {
+                $node = random_int(0, 0xffffffffffff);
+                if (false === apcu_store('__symfony_uuid_node', $node)) {
+                    goto static_node;
+                }
+            }
+        } else {
+            static_node:
+            static $node;
+            if (null === $node) {
+                $node = random_int(0, 0xffffffffffff);
+            }
+        }
```

Comment:
Wouldn't it be faster to do avoid calling `function_exists` and `apcu_fetch` each time we generate a uuid?
```php
        static $node;
        if (null === $node) {
            if (\function_exists('apcu_fetch')) {
                $node = apcu_fetch('__symfony_uuid_node');
                if (false === $node) {
                    $node = random_int(0, 0xffffffffffff);
                    apcu_store('__symfony_uuid_node', $node);
                }
            } else {
                $node = random_int(0, 0xffffffffffff);
        }

```

Reply:
Much better (and bye bye my goto :cry: )

Reply:
Note: I also fixed the support for 32-bits Sytems